File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/FungiBeast_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class FungiBeast_Model extends MonsterInfo{
    public static Id = "FungiBeast"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(24, 28);
        else
            return Mathf.RandomRange(22, 28);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/fungi/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/fungi/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/fungi/skeleton.png"));
    }

    get Name(): string {
        return "真菌兽";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/FungiBeast_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class FungiBeast_Model : MonsterInfo
    {
        public static string Id = "FungiBeast";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(24, 28) : Random.Range(22,28);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/fungi/skeleton.atlas",
                    "Monsters/theBottom/fungi/skeleton.json", "Monsters/theBottom/fungi/skeleton.png");
            }
        }

        public override string Name => "真菌兽";

        public FungiBeast_Model(Vector2 pos) : base(pos)
        {
        }
    }
}